Binder and Notebooks documentation improvements (#421)
EGI-Federation/documentation: content/en/users/dev-env/binder/_index.md

type: docs
weight: 20
description: >
  Custom reproducible computing environments for notebooks
---

## What is it?

Binder allows the re-creation of a custom computing environment for reproducible
execution of notebooks (and potentially many other types of applications). Users

> who create their own notebooks in the [EGI Notebooks](https://notebooks.egi.eu)

Reviewer: Shouldn't this link to the Notebooks users doc instead (that page contains a link to the actual service, instead)?

Author: yes, this was changed with Gergely. But I do agree it makes sense to move it to the docs instead. Fixing it

aliases:
  - /users/notebooks/kernels/binder
description: >
  Custom reproducible computing environments for notebooks
---

## What is it?

Binder allows the re-creation of a custom computing environment for reproducible
execution of notebooks (and potentially many other types of applications). Users
who create their own notebooks in the [EGI Notebooks](../notebooks/)
to analyze data can easily create a shareable link for those notebooks in the
form of a GitHub repository. Based on this link, anyone can then reproduce the
same data analysis using the link in the
[EGI Binder](https://binder.notebooks.egi.eu) service.

The service builds on [BinderHub](https://github.com/jupyterhub/binderhub), an
Open Source tool that allows to build docker images from a Git repository and

> then make it available through your browser.

Reviewer 2: ```suggestion
then makes them available through your browser.
```

aliases:
  - /users/notebooks/kernels/binder
description: >
  Custom reproducible computing environments for notebooks
---

## What is it?

Binder allows the re-creation of a custom computing environment for reproducible
execution of notebooks (and potentially many other types of applications). Users
who create their own notebooks in the [EGI Notebooks](../notebooks/)
to analyze data can easily create a shareable link for those notebooks in the
form of a GitHub repository. Based on this link, anyone can then reproduce the
same data analysis using the link in the
[EGI Binder](https://binder.notebooks.egi.eu) service.

> The service builds on [BinderHub](https://github.com/jupyterhub/binderhub), an
> Open Source tool that allows to build docker images from a Git repository and
> then make it available from JupyterHub.

Reviewer 2: ```suggestion
The service builds on [BinderHub](https://github.com/jupyterhub/binderhub), an
Open Source tool allowing to build docker images from a Git repository, and
then making them available via JupyterHub-based services like
[EGI Notebooks](../notebooks).
```

Reviewer 2: Not sure about this, but I felt it would be good to be more concrete and link with our own service, if appropriate.

Author: I did a reword here, take a look

EGI Binder offers a service similar to the publicly accessible

> [mybinder.org](https://mybinder.org) site. However EGI Binder has the following

Reviewer 2: ```suggestion
[mybinder.org](https://mybinder.org) site. However, EGI Binder has the following
```

additional features:

- Access with academic user accounts: login via [Check-in](../../aai/check-in)
  that’s connected to eduGAIN and social media accounts.
- Access to scalable storage: selected storage spaces of EGI DataHub are
  directly available under the `datahub` folder simplifying the access to shared
  data from Binder notebooks.

> - Guaranteed capacity: environments are guaranteed 2GB of RAM and can reach 4GB

Reviewer 2: ```suggestion
- Guaranteed capacity: environments have 2GB of RAM guaranteed and can reach 4GB
```

- Community Binder environments: User communities can have their customized
  Binder service instance from EGI, with extra features as requested (such as
  access to GPUs, integration with community specific data repositories and
  services). EGI offers consultancy and support the setup of these instances,
  and provides operational oversight for them.

## Reproducible research

Binder facilitates the sharing and reproducibility of digital data analysis:

1. Users can define their computational analysis in the
   [EGI Notebooks](../notebooks/) service.
1. Once the notebook is ready for publishing, it can be shared in a GitHub
   repository.
1. Optionally, users can use the Zenodo-GitHub integration for generating DOIs
   that can be cited in publications and can be discovered by fellow researchers
1. Anyone can use the link to the GitHub repository or Zenodo DOI to reproduce
   the computational analysis in [EGI Binder](https://binder.notebooks.egi.eu).

![Reproducible research flow](reproduce-flow-binder.png)

## Access to the service

[EGI's Binder](https://binder.notebooks.egi.eu/) has the same access conditions

> as the [centrally operated Notebooks service from EGI](../../#service-modes).

Author: ```suggestion
as the [centrally operated Notebooks service from EGI](../notebooks/#service-modes).
```

Before using the service, you need to have an EGI account and be a member of
[one of the supported resource pools](../notebooks/#notebooks-for-researchers)
(alias Virtual Organisations). Follow the instructions on the

> [EGI Binder login page](https://binder.notebooks.egi.eu) for access

Reviewer 2: ```suggestion
[EGI Binder login page](https://binder.notebooks.egi.eu) to access.
```

environments; installing Python, R and Julia environments; installing additional
OS packages; and even complete custom Dockerfiles to bring any application to
the system. The code repository can be hosted on popular git hosting platforms
like GitHub and GitLab and can also be referenced with a DOI from Zenodo,
FigShare or Dataverse. You can learn more on the configuration of your
repository with Binder at the
[Binder user documentation](https://mybinder.readthedocs.io/en/latest/)

You can start by forking the
[EGI-Federation/binder-example GitHub repository](https://github.com/EGI-Federation/binder-example)
for creating your own reproducible environment. To run this directly on EGI's
Binder click on the button below:

[![Binder](https://binder.notebooks.egi.eu/badge_logo.svg)](https://binder.notebooks.egi.eu/v2/gh/EGI-Federation/binder-example/HEAD)

You can create such link to share your notebooks from the Binder interface, as
shown in the screenshot below, you can copy the URL shown when the building is
in progress:

![Binder link](binder-link.png)

The [binder examples](https://github.com/binder-examples) organization on GitHub

> contains more sample repositories for common configurations that can help you to
> get started.

Reviewer 2: ```suggestion
contains more sample repositories for common configurations that can help you
getting started.
```

## Accessing data

Your notebooks running in Binder have outgoing internet connectivity, so you can

> connect to external services to bring data in for analysis or deposit the

Reviewer 2: ```suggestion
connect to external services to bring data in for analysis or deposing the
```
